### DISCUSSION TITLE
Maintaining client sessions across subdomains on the same host

### DISCUSSION THREAD

**Discussion by aymanimtyaz-goswift** [encode/httpx#2952] [Ideas] [answered]
Is this something that is possible? From my basic understanding of TCP and HTTP, subdomains have the same host and as such, the underlying TCP connection can be reused across subdomains.
Is there a way to do this using httpx client sessions?

**Comment by karpetrosyan** [comment, marked as answer]
Subdomains do not share the same host. The owner of domain X also owns all of its subdomains, but they can refer to different IPs, so you cannot use the same TCP connection.

Maybe you are talking about reusing the connections on the same host, which is supported by httpx.

**Reply by aymanimtyaz-goswift** [reply]
Thanks for your answer!
